packages/card/index.js

```diff
@@ -29,42 +57,24 @@ class WarpCard extends WarpElement {
     `,
   ];
 
-  static properties = {
-    selected: { type: Boolean, reflect: true },
-    flat: { type: Boolean },
-    clickable: { type: Boolean },
-  };
-
-  constructor() {
-    super();
-    this.selected = false;
-    this.flat = false;
-    this.clickable = false;
-  }
-
   get _containerClasses() {
-    return fclasses({
-      [ccCard.card]: true,
-      [ccCard.cardShadow]: !this.flat,
-      [ccCard.cardSelected]: !this.flat && this.selected,
-      [ccCard.cardFlat]: this.flat,
-      [this.selected ? ccCard.cardFlatSelected : ccCard.cardFlatUnselected]: this.flat,
-    });
+    return classNames([
+      ccCard.base,
+      this.flat ? ccCard.flat : ccCard.shadow,
+      this.selected ? (this.flat ? ccCard.flatSelected : ccCard.selected) : this.flat && ccCard.flatUnselected,
+    ]);
   }
 
   get _outlineClasses() {
-    return fclasses({
-      [ccCard.cardOutline]: true,
-      [this.selected ? ccCard.cardOutlineSelected : ccCard.cardOutlineUnselected]: true,
-    });
+    return classNames([ccCard.outline, this.selected ? ccCard.outlineSelected : ccCard.outlineUnselected]);
   }
 
-  get uuButton() {
-    return html`<button class="${ccCard.a11y}" aria-pressed="${this.selected}" tabindex="-1">Velg</button>`;
-  }
+  get _interactiveElement() {
+    const renderButton = () =>
+      html`<button class="${ccCard.a11y}" aria-pressed="${this.selected}" tabindex="-1">${this.buttonText}</button>`;
+    const renderSpan = () => html`<span role="checkbox" aria-checked="true" aria-disabled="true"></span>`;
 
-  get uuSpan() {
-    return html`<span role="checkbox" aria-checked="true" aria-disabled="true"></span>`;
+    return this.clickable ? renderButton() : !this.clickable && this.selected ? renderSpan() : '';
```

Comment:
The `: !this.clickable` should be redundant.

This instead?
`return this.clickable ? renderButton() : this.selected ? renderSpan() : '';`

Reply:
ah good catch! 😄